### PR TITLE
Improve documentation for backends

### PR DESCRIPTION
**Pull request by LaurinFischer**
### Summary
This PR improves the distinction between the general-purpose simulator backends for spin and fermionic circuits, and the  mock backends that simulates a real device with sophisticated configuration. 

### Details and comments
Besides slight changes to some docstrings, the `FermionSimulator` and `SpinSimulator` general backends are now also added to the default backends of the provider, so that users don't miss these backends when they check the provider for available backends in their workflow.

### CODE REVIEW

File: qiskit_cold_atom/providers/collective_spin_backend.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""Backend to simulate collective spin experiments."""
+"""Mock backend to simulate collective spin experiments."""
```
Comment (eggerdj) [inline qiskit_cold_atom/providers/collective_spin_backend.py:13]:
```suggestion
"""Backend to emulate collective spin experiments."""
```

File: qiskit_cold_atom/providers/collective_spin_backend.py [hunk 2]
```diff
@@ -20,7 +20,8 @@
 
 
 class CollectiveSpinSimulator(SpinSimulator):
-    """Simulator backend of a collective spin system of trapped BECs in optical tweezers."""
+    """Mock backend of a spin device with collective spins of trapped BECs in optical tweezers.
+    For a general spin simulator backend, use the SpinSimulator base class."""
```
Comment (eggerdj) [inline qiskit_cold_atom/providers/collective_spin_backend.py:24]:
```suggestion
    """Emulator backend of a spin device with collective spins of trapped BECs in optical tweezers.
    
    This backend will define a configuration with a set of supported gates and a line coupling map.
    At instantiation users can chose the length of the coupling map which by default is three. This 
    backend is intended to be a realistic representation of a collective spin experiment with BECs. 
    For a general spin simulator backend, use the SpinSimulator base class.
    """
```

File: qiskit_cold_atom/providers/fermionic_tweezer_backend.py [hunk 3]
```diff
@@ -10,13 +10,14 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""Backend to simulate fermionic tweezer experiments."""
+"""Mock backend to simulate fermionic tweezer experiments."""
```
Comment (eggerdj) [inline qiskit_cold_atom/providers/fermionic_tweezer_backend.py:13]:
```suggestion
"""Backend to emulate fermionic tweezer experiments."""
```

File: qiskit_cold_atom/providers/fermionic_tweezer_backend.py [hunk 4]
```diff
@@ -10,13 +10,14 @@
...
 
 from qiskit_cold_atom.fermions.fermion_simulator_backend import FermionSimulator
 
 
 class FermionicTweezerSimulator(FermionSimulator):
-    """simulator backend of a fermionic hardware with four tweezer sites that uses two spin species"""
+    """Mock backend of a fermionic device with four tweezer sites that uses two spin species.
+    For a general fermionic simulator backend, use the FermionSimulator base class."""
```
Comment (eggerdj) [inline qiskit_cold_atom/providers/fermionic_tweezer_backend.py:20]:
```suggestion
    """Emulator backend of a fermionic device with n tweezer sites that uses two spin species.
    
    This backend will define a configuration with a set of supported gates and a line coupling map.
    At instantiation users can chose the length of the coupling map which by default is four sites.
    This backend is intended to be a realistic representation of a ferminoic tweezer experiment.
    For a general fermionic simulator backend, use the FermionSimulator base class.
    """
```

**Review by eggerdj — CHANGES_REQUESTED** [review]
Looking good. I think we can say a bit in the doc.

File: qiskit_cold_atom/providers/fermionic_tweezer_backend.py (hunk 4 above)
Comment (LaurinFischer) [reply qiskit_cold_atom/providers/fermionic_tweezer_backend.py]: Yea okay, I didn't wanna make this too long but I agree the added level of detail is important. Thanks!

**Review by eggerdj — APPROVED** [review]
LGTM

**Pull request merged**